src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/action/start/TransportStartTransformAction.kt

```diff
@@ -0,0 +1,150 @@
+                    if (transform.metadataId != null) {
+                        getTransformMetadata(transform, actionListener)
+                    } else {
+                        actionListener.onResponse(AcknowledgedResponse(true))
+                    }
+                } else {
+                    actionListener.onResponse(AcknowledgedResponse(false))
+                }
+            }
+
+            override fun onFailure(e: Exception) {
+                actionListener.onFailure(ExceptionsHelper.unwrapCause(e) as Exception)
+            }
+        })
+    }
+
+    private fun getTransformMetadata(transform: Transform, actionListener: ActionListener<AcknowledgedResponse>) {
+        val req = GetRequest(INDEX_MANAGEMENT_INDEX, transform.metadataId).routing(transform.id)
+        client.get(req, object : ActionListener<GetResponse> {
+            override fun onResponse(response: GetResponse) {
+                if (!response.isExists || response.isSourceEmpty) {
+                    // If there is no metadata doc then the runner will instantiate a new one
+                    // in FAILED status which the user will need to retry from
+                    actionListener.onResponse(AcknowledgedResponse(true))
+                } else {
+                    val metadata = response.sourceAsBytesRef?.let {
+                        val xcp = XContentHelper.createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, it, XContentType.JSON)
+                        xcp.parseWithType(response.id, response.seqNo, response.primaryTerm, TransformMetadata.Companion::parse)
+                    }
+                    if (metadata == null) {
+                        // If there is no metadata doc then the runner will instantiate a new one
+                        // in FAILED status which the user will need to retry from
+                        actionListener.onResponse(AcknowledgedResponse(true))
+                    } else {
+                        updateTransformMetadata(transform, metadata, actionListener)
+                    }
+                }
+            }
+
+            override fun onFailure(e: Exception) {
+                actionListener.onFailure(ExceptionsHelper.unwrapCause(e) as Exception)
+            }
+        })
+    }
+
+    private fun updateTransformMetadata(transform: Transform, metadata: TransformMetadata, actionListener: ActionListener<AcknowledgedResponse>) {
+        val now = Instant.now().toEpochMilli()
+        val updatedStatus = when (metadata.status) {
+            TransformMetadata.Status.FINISHED, TransformMetadata.Status.STOPPED -> TransformMetadata.Status.STARTED
+            TransformMetadata.Status.STARTED, TransformMetadata.Status.INIT ->
+                return actionListener.onResponse(AcknowledgedResponse(true))
+            TransformMetadata.Status.FAILED -> TransformMetadata.Status.FAILED // What should happen here?
```

Comment:
I think for now we can just move to `STARTED`. When we have incremental transforms we can see if a new status is required